lib/cosmos/interfaces/tcpip_client_interface.rb

```diff
@@ -42,19 +45,24 @@ def initialize(hostname,
       @write_raw_allowed = false unless @write_port
     end
 
-    # Connects the {StreamProtocol} to a {TcpipClientStream} by passing the
+    # Connects the {TcpipClientStream} by passing the
     # initialization parameters to the {TcpipClientStream}.
     def connect
-      stream = TcpipClientStream.new(
-        @hostname,
-        @write_port,
-        @read_port,
-        @write_timeout,
-        @read_timeout)
-      stream.raw_logger_pair = @raw_logger_pair
-      @stream_protocol.connect(stream)
+      @stream = TcpipClientStream.new(@hostname,
+                                      @write_port,
+                                      @read_port,
+                                      @write_timeout,
+                                      @read_timeout)
+      @stream.raw_logger_pair = @raw_logger_pair
+      @stream.connect
```

Comment:
This stream instance variable is automatically used by the stream protocol module as you saw earlier. Obviously we have to be consistent in our naming for all that to work.